view_as_real doesn't work for all backends since it relies on strides.

ailzhang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47018 view_as_real doesn't work for all backends since it relies on strides.**

Differential Revision: [D24607340](https://our.internmc.facebook.com/intern/diff/D24607340)

Commit:
view_as_real doesn't work for all backends since it relies on strides.

[ghstack-poisoned]

albanD commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -283,7 +283,7 @@
   variants: function
   dispatch:
-    DefaultBackend: view_as_real
+    CPU, CUDA: view_as_real
```
I guess the same is true for view_as_complex just below?

ailzhang replied:
ah nice catch!

Commit:
Update on "view_as_real doesn't work for all backends since it relies on strides."


Differential Revision: [D24607340](https://our.internmc.facebook.com/intern/diff/D24607340)

[ghstack-poisoned]

albanD reviewed (approved):
LGTM thanks for the fix!

anjali411 commented:
should similar changes be made for other view functions as well? For example, `torch.t()` (which currently dispatches to `DefaultBackend`)

albanD commented:
I would say yes.
Especially because we can easily test this behavior by trying to call this function with a sparse Tensor.
